core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1094,11 +1111,11 @@ public IndexShardCacheEntity(IndexShard indexShard, QueryPhase queryPhase, Searc
              * results.*/
             final int expectedSizeInBytes = 512;
```

Comment:
> BytesStreamOutput is pretty good at resizing I think we are fine here

Maybe I'm missing something but it looks like we never trim it before we put it in the cache.

> The other question is if we should raise the overall size of the case since we now also cache field stats?

Yeah, maybe some? You'd get one entry per field on the node. If we have them as .5k then that is 100k for 200 fields. I'm not sure how much larger to make it?